notes/lecture5.tex

```diff
@@ -0,0 +1,141 @@
+
+\subsubsection{Low-rank matrix completion}
+Suppose we have a partially observable matrix $Y$ and we would like to 
+find its completion form projected in a nuclear norm ball. Formally we have the objective function
+\[
+\min_{X\in B_*}\frac{1}{2}\|Y-X \odot O\|_F^2
+\]
+And calculate the gradient of this function we will have
+\[
+\nabla f(X) = Y-X \odot O
+\]
+We can use projected gradient descent to solve this problem but it is more efficient to use Frank-Wolfe algorithm. We need to solve the linear optimization oracle
+\[
+\bar{X}_t\in \argmin_{X\in{B_*}}f(X_t)+\nabla f(X_t)^\trans(X-X_t)
+\]
+This will lead to a rank-1 matrix which is decomposed from $-\nabla f(X_t)$. This can be derived from \lemmaref{lemma}. Now we can have the update rule for the conditional gradient as
+\[
+X_{t+1} = X_t + \eta_t(-u_1v_1^\trans - X_t)
+\]
+where $u_1$ and $v_1$ are the top left and right singular vectors.\\ \\
+
+\begin{lemma}
+\lemmalabel{lemma}
+\[
+\min_{X\in B_*}\langle\nabla f(X_t)^\trans,X\rangle,\quad B_*=\{X|\|X\|_*\leqslant 1\}
+\]
+The optimal result is \[-\|\nabla f(X_t)\|\] with $X=-u_1v_1^\trans$ being one of the X that minimize the function where
+$u_1$ and $v_1$ are left and right singular vectors corresponding to the max singular value, $\|\nabla f(X_t)\|=\max_{\|X\|_*\leqslant 1}<\nabla f(X_t),X>$.\\
+\end{lemma}
+
+\begin{proof} [Proof of \lemmaref{lemma}.]
+Since the dual norm of a nuclear norm is operator norm,
+\[
+\|Y\| = \max_{\|X\|_*\leq 1}\langle Y, X\rangle
+\]we can easily see that the optimal result of the objective function in \lemmaref{lemma} being
+\[
+-\|Y\| = -\nabla f(X_t)=-\sigma_{\max}.
+\]
+Then we show by Singular value decompasition to prove that $X = -u_1v_1^\trans$ is one of the solutions that minimize the function.
+From SVD, $\nabla f(X_t) = U\Sigma V^\trans $, then $\Sigma = U^\trans YV$, we set $X=-u_1v_1^\trans$, thus
+\[
+\langle\nabla f(X_t), X\rangle = \nabla f(X_t)^\trans X = -V\Sigma U^\trans u_1v_1^\trans= -\sigma_1
+\]
+\end{proof}
+According to the lemma, we can see that only the leading singular value and corresponding vectors are used to deprive the argmin X. Thus we can do the rank-1 approximation of this function which makes no difference to final result.\\ \\
+Power method works well in approximate the leading singular value and corresponding left and right vetors.\\
+Here is the introduction of the process of SVD power method.
+\begin{itemize}
+\item $w_k=Av_{k-1}, \alpha_k=\|w_k\|, u_k={\alpha_k}^{-1}w_k$
+\item $z_k=A^\trans, \beta_k = \|z_k\|, v_k=\beta_kz_k$
+\end{itemize}
+$k=1, 2, 3, 4, 5,... , \sigma_{max}=\beta_k, v_k=\delta_{2k}\sum\limits_{i=1}^r \sigma_j ^{2k}y_jv_j, u_k=\delta_{2k+1}\sum\limits_{i=1}^r \sigma_j ^{2k+1}y_jv_j$\\
```

Comment:
I don't know what you're describing here, but it's not what we did in class. Please look at the code for a proper description of the algorithm.